sklearn/neighbors/base.py

```diff
@@ -258,6 +258,12 @@ def _fit(self, X):
                     "Expected n_neighbors > 0. Got %d" %
                     self.n_neighbors
                 )
+            else:
+                if not np.issubdtype(type(self.n_neighbors), np.integer):
+                    raise TypeError(
+                        "n_neighbors does not take %s value, "
+                        "enter integer value" %
+                        str(type(self.n_neighbors)).split("'")[1])
```

Comment:
No, this is the wrong fix. Not all types will look like that.

Reply:
Hard coding the expected message fails the test, I tried both <class 'float'> and  <type 'float'>. What can be the fix then?